python/pyspark/tests.py

```diff
@@ -1686,7 +1686,7 @@ def test_single_script_on_cluster(self):
         # this will fail if you have different spark.executor.memory
         # in conf/spark-defaults.conf
         proc = subprocess.Popen(
-            [self.sparkSubmit, "--master", "local-cluster[1,1,512]", script],
+            [self.sparkSubmit, "--main", "local-cluster[1,1,512]", script],
```

Comment:
I'm all for the spirit of this diff -- but this repo hasn't been updated in 6 years, and the version of spark it uses (1.3.0) uses master as an argument. Changing this "--master" to "--main" will break it. Some of these changes could probably be made (the methods defined in this module) but given there isn't a lot of test coverage and this isn't an active repo, I'd be hesitant to merge this. But I'll defer to @ryleg on this one.